src/OrbitVulkanLayer/VulkanLayerController.h

```diff
@@ -181,8 +180,9 @@ class VulkanLayerController {
 
     // Need to call vkCreateInstance down the chain to actually create the
     // instance, as we need it to be alive in the create instance dispatch table.
-    auto create_device_function = absl::bit_cast<PFN_vkCreateDevice>(
-        next_get_instance_proc_addr_function(VK_NULL_HANDLE, "vkCreateDevice"));
+    auto create_device_function =
+        absl::bit_cast<PFN_vkCreateDevice>(next_get_instance_proc_addr_function(
+            dispatch_table_.GetInstance(physical_device), "vkCreateDevice"));
```

Comment:
> per API, this call will return
a nullptr and the subsequent call to the function pointer will
fail. While most ICDs and layers will return the right function
pointer anyways, Renderdoc returns null in that case.

Maybe more of a curiosity, but: Does this mean that when passing `VK_NULL_HANDLE` then nullptr should be returned, while the right function should be returned when passing an instance? Or does it mean that nullptr _can_ be returned when passing `VK_NULL_HANDLE` (basically undefined behavior in this case)?

Reply:
It's undefined.
https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/vkGetInstanceProcAddr.html#_description

Reply:
And thanks!